homeassistant/components/vicare/button.py

```diff
@@ -0,0 +1,150 @@
+"""Viessmann ViCare sensor device."""
+from __future__ import annotations
+
+from contextlib import suppress
+from dataclasses import dataclass
+import logging
+
+from PyViCare.PyViCareUtils import (
+    PyViCareInvalidDataError,
+    PyViCareNotSupportedFeatureError,
+    PyViCareRateLimitError,
+)
+import requests
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ENTITY_CATEGORY_CONFIG
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ViCareRequiredKeysMixin
+from .const import DOMAIN, VICARE_API, VICARE_DEVICE_CONFIG, VICARE_NAME
+
+_LOGGER = logging.getLogger(__name__)
+
+BUTTON_DHW_ACTIVATE_ONETIME_CHARGE = "activate_onetimecharge"
+
+
+@dataclass
+class ViCareButtonEntityDescription(ButtonEntityDescription, ViCareRequiredKeysMixin):
+    """Describes ViCare button sensor entity."""
+
+
+BUTTON_DESCRIPTIONS: tuple[ViCareButtonEntityDescription, ...] = (
+    ViCareButtonEntityDescription(
+        key=BUTTON_DHW_ACTIVATE_ONETIME_CHARGE,
+        name="Activate one-time charge",
+        icon="mdi:shower-head",
+        entity_category=ENTITY_CATEGORY_CONFIG,
+        device_class=ButtonDeviceClass.RESTART,
```

Comment:
Is this really a proper device class for this action? Does the button restart the device?

Reply:
good point. While it does "restart" the domestic hot water heating what's meant with  ButtonDeviceClass.RESTART is probably a restart of a device. I'll rather not set it then.